src/cmd/build.js

```diff
@@ -124,11 +130,42 @@ async function defaultPackageCreator({
   const packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   const extensionPath = path.join(artifactsDir, packageName);
-  const stream = createWriteStream(extensionPath);
+  let destStat;
+  try {
+    destStat = await fs.stat(extensionPath);
+  } catch (error) {
+    destStat = false;
+    throw error;
```

Comment:
@harikishen it looks like the existent tests are currently failing because fs.stat raises an error when the "extensionPath" does not exist (it raises an error object with code 'ENOENT', e.g. like documented in the following nodejs doc pages: https://nodejs.org/api/errors.html#errors_common_system_errors and https://nodejs.org/api/fs.html#fs_fs_stat_path_callback) and so we should throw the caught error only if the error code is not 'ENOENT' (which is an expected error when we are building a file to a path that doesn't exist yet).

Applying the following change should fix the existent tests:
```diff
diff --git a/src/cmd/build.js b/src/cmd/build.js
index 125eed7..6d123d2 100644
--- a/src/cmd/build.js
+++ b/src/cmd/build.js
@@ -11,7 +11,7 @@ import {zipDir} from '../util/zip-dir';
 import getValidatedManifest, {getManifestId} from '../util/manifest';
 import {prepareArtifactsDir} from '../util/artifacts';
 import {createLogger} from '../util/logger';
-import {UsageError} from '../errors';
+import {UsageError, isErrorWithCode} from '../errors';
 import {
   createFileFilter as defaultFileFilterCreator,
   FileFilter,
@@ -135,7 +135,9 @@ async function defaultPackageCreator({
     destStat = await fs.stat(extensionPath);
   } catch (error) {
     destStat = false;
-    throw error;
+    if (!isErrorWithCode('ENOENT', error)) {
+      throw error;
+    }
   }
   const destinationFileExists = destStat && destStat.isFile();
```